devito/finite_differences/tools.py

```diff
@@ -325,3 +309,16 @@ def make_shift_x0(shift, ndim):
             raise ValueError("ndim length must be equal to 1 or 2")
     raise ValueError("shift parameter must be one of the following options: "
                      "None, float or tuple with shape equal to %s" % (ndim,))
+
+
+def process_weights(weights, expr):
+    if weights is None:
+        return 0, None
+    elif isinstance(weights, Function):
+        wdim = {d for d in weights.dimensions if d not in expr.dimensions}
```

Comment:
potential nitpicking and therefore ignorable:

is it worth introducing a `SymbolicWeights` subclass (can't call it just `Weights` because it'd be the same name as `finite_differences/differentiable.py::Weights`) and add a few properties such as `.dimension` ? or is this impossible because you need `expr`?